docs/BacwardCompatibility.md

```diff
@@ -0,0 +1,239 @@
+# Fybrik Versions, Compatibility Requirements and Implementation Plan
+
+## Disclaimer
+This document was created according to the Fybrik release v1.1.0, future releases may require updates code pointers
+
+## What is Compatibility
+Before talking about how to provide compatability, it is worthwhile to clarify what we mean by compatibility. 
+Fybrik is required to provide forwards and backwards compatibility in its behaviour. Compatibility is hard, especially 
+handling issues around rollback-safety. 
+Let's try to define compatability in this way:
+- Existing clients need not be aware of a new change in order to continue to function as they did previously, 
+even when the change is in use.
+- Any Fybrik operation that succeeded before the new change must succeed after the change.
+- Any Fybrik operation that does not use the new change must behave the same as it did before the change.
+- Any Fybrik operation that use the new change must not cause problems (e.g. crash or degrade behavior) when issued 
+against Fybrik objects, e.g. FybrikModules, Plotters, Blueprints and FybrikApplications created before the functionality was added.
+The objects should be smoothly promoted to the new version (if required).
+- It must be possible to round-trip Fybrik version changes (convert to different API versions and back) with no loss of 
+information (Should we support downgrade ?)
+
+*Note:* Fybrik allows to its users to [redefine most of the components](https://fybrik.io/v1.1/concepts/taxonomy/#taxonomy-contributors) 
+by using a custom taxonomy. If any of the components were redefined, it will be the user responsibility to update its 
+custom taxonomy files and use them during **each** 
+fybrik [upgrade](https://fybrik.io/v1.1/tasks/custom-taxonomy/#deploy-fybrik-with-custom-taxonomy). 
+
+## Fybrik Version Policy
+Fybrik uses [semantic versioning model](https://semver.org), which is the module adopted by a wide range of projects,
+e.g. Kubernetes, goLang, Helm and many others.
+
+Fybrik versions are of the form "vX.Y.Z" where X is the major version number, Y is the minor version number
+and Z is the patch release number.
+- Major Releases - released less frequently, indicate that **backward-incompatible public API changes**. This release
+  carries no guarantee that it will be backward compatible with preceding major versions.
+- Minor Releases - indicate **backward-compatible public API changes**. This release guarantees backward compatibility
+  and stability. The backward compatibility policy is defined in a separate document [ADD link] but in general, Fybrik
```

Comment:
Is there a link? Will be added in another PR?

Reply:
fixed.